Gulpfile.ts

```
@@ -0,0 +1,1125 @@
+        fs.writeFileSync(builtGeneratedDiagnosticMessagesJSON, fs.readFileSync(generatedDiagnosticMessagesJSON));
+    }
+    done();
+});
+
+gulp.task("generate-diagnostics", "Generates a diagnostic file in TypeScript based on an input JSON file", [diagnosticInfoMapTs]);
+
+
+const servicesFile = path.join(builtLocalDirectory, "typescriptServices.js");
+const standaloneDefinitionsFile = path.join(builtLocalDirectory, "typescriptServices.d.ts");
+const nodePackageFile = path.join(builtLocalDirectory, "typescript.js");
+const nodeDefinitionsFile = path.join(builtLocalDirectory, "typescript.d.ts");
+const nodeStandaloneDefinitionsFile = path.join(builtLocalDirectory, "typescript_standalone.d.ts");
+
+gulp.task(builtLocalCompiler, false, [servicesFile], () => {
+    const localCompilerProject = tsc.createProject("src/compiler/tsconfig.json", {typescript: require("./built/local/typescript.js")});
+    let result: NodeJS.ReadWriteStream = localCompilerProject.src()
+        .pipe(sourcemaps.init())
+        .pipe(tsc(localCompilerProject));
+    if (!useDebugMode) {
+        result = result.pipe(insert.prepend(fs.readFileSync(copyright)));
+    }
+    return result.pipe(sourcemaps.write("."))
+        .pipe(gulp.dest(builtLocalDirectory));
+});
+
+gulp.task(servicesFile, false, ["lib", "generate-diagnostics"], (done) => {
+    const settings: tsc.Settings = getCompilerSettings({
+        declaration: true,
+        preserveConstEnums: true,
+        removeComments: false,
+        noResolve: false,
+        stripInternal: true,
+        outFile: servicesFile
+    }, /*useBuiltCompiler*/ false);
+    const {js, dts} = gulp.src(servicesSources)
+        .pipe(sourcemaps.init())
+        .pipe(tsc(settings));
+    let result: NodeJS.ReadableStream = js;
+    if (!useDebugMode) {
+        result = result.pipe(insert.prepend(fs.readFileSync(copyright)));
+    }
+    result.pipe(sourcemaps.write("."))

```

I had actually been merging them originally but I had some random issue with errors getting swallowed. I probably need to merge in the sequence of streams so that errors get merged at all stages?